acapy_agent/wallet/keys/manager.py

```diff
@@ -66,6 +75,20 @@ def __init__(self, session: ProfileSession):
         self.session: ProfileSession = session
         self.wallet: BaseWallet = session.inject(BaseWallet)
 
+    async def idempotent_resolve_and_store_from_kid(self, kid: str):
+        """Fetch key if exists, otherwise resolve and store it"""
```

Comment:
```suggestion
    async def resolve_and_bind_kid(self, kid: str):
        """Fetch key if exists, otherwise resolve and store it. This function is idempotent."""
```

Reply:
Since its not "storing" a key, I think the term "bind" is preferable. It's binding a key id to a key which already exists.